src/modules/cmdpal/Microsoft.CmdPal.UI/App.xaml.cs

```diff
@@ -72,6 +73,8 @@ private static ServiceProvider ConfigureServices()
 
         // Root services
         services.AddSingleton(TaskScheduler.FromCurrentSynchronizationContext());
+        services.AddSingleton(DispatcherQueue.GetForCurrentThread());
+        services.AddSingleton<IIconCacheService, IconCacheService>();
```

Comment:
Rather than passing the DispatcherQueue here (could lead to abuse), just provide the constructed version of the service (or use the factory pattern) to construct the IconCacheService here so that you can pass the DispatcherQueue reference from here as well.

```suggestion
        services.AddSingleton<IIconCacheService, IconCacheService>(new(DispatcherQueue.GetForCurrentThread()));
```

Basically, stuff in the View layer should just have access to the DispatcherQueue whenever it needs anyway, or can just be pulled in directly. The main advantage of the services collection here is to bridge the View and ViewModels for things that need to live in the View (like the UI thread via the SynchronizationContext [the .NET framework agnostic thread handle] vs. the DispatcherQueue which is specific to Windows, so we don't want to leak that into our VMs).